File: A-Set/0 - 100/71 - A Way Too Long Words.py
#!/usr/bin/env python
integer = int(input())

for x in range(integer):
  counter = 0
  text = str(input())
  text = list(text)
  length = len(text)

  if len(text) <= 10:
    print(''.join(map(str, text)))
  else:
    for i in text[1:-1]:
      counter += 1
      first_char = text[0]
      last_char = text[-1]
    print(first_char + str(counter) + last_char)

File: codeforces/A-Set/0 - 100/71 - A Way Too Long Words.c
#include <stdio.h>
#include <string.h>

int main() {
  int totalWord, i, k;
  char first_char, last_char;
  char string[100000];

  scanf("%d", &totalWord);

  for(i = 0; i < totalWord; i++) {
    int counter = 0;
    scanf("%s", string);
    int length = strlen(string);
    
    if(length <= 10) {
      printf("%s\n", string);
    }
    else {
      for(k = 0; k < length - 2; k++) {
        first_char = string[0];
        last_char = string[length-1];
        counter++;
      }
      printf("%c%d%c\n", first_char, counter, last_char);
    }
  }
}